tests/model_serving/model_server/kueue/test_kueue_isvc_raw.py

```diff
@@ -0,0 +1,123 @@
+)
+max_replicas = 2
+
+
+@pytest.mark.rawdeployment
+@pytest.mark.parametrize(
+    "unprivileged_model_namespace, kueue_kserve_serving_runtime, kueue_raw_inference_service, "
+    "kueue_cluster_queue_from_template, kueue_resource_flavor_from_template, kueue_local_queue_from_template",
+    [
+        pytest.param(
+            {"name": "kueue-isvc-raw-test", "add-kueue-label": True},
+            RunTimeConfigs.ONNX_OPSET13_RUNTIME_CONFIG,
+            {
+                "name": "kueue-isvc-raw",
+                "min-replicas": min_replicas,
+                "max-replicas": max_replicas,
+                "labels": {"kueue.x-k8s.io/queue-name": local_queue_name},
+                "deployment-mode": KServeDeploymentType.RAW_DEPLOYMENT,
+                "model-dir": "test-dir",
+                "model-version": ModelVersion.OPSET13,
+                "resources": isvc_resources,
+            },
+            {
+                "name": cluster_queue_name,
+                "resource_flavor_name": resource_flavor_name,
+                "cpu_quota": cpu_quota,
+                "memory_quota": memory_quota,
+                "namespace_selector": {},
+            },
+            {"name": resource_flavor_name},
+            {"name": local_queue_name, "cluster_queue": cluster_queue_name},
+        )
+    ],
+    indirect=True,
+)
+class TestKueueInferenceServiceRaw:
+    """Test inference service with raw deployment"""
+
+    def test_kueue_inference_service_raw(
+        self,
+        admin_client,
+        kueue_cluster_queue_from_template,
+        kueue_resource_flavor_from_template,
+        kueue_local_queue_from_template,
+        kueue_raw_inference_service,
+        kueue_kserve_serving_runtime,
+    ):
+        """Test inference service with raw deployment"""
+        labels = [
+            create_isvc_label_selector_str(
+                isvc=kueue_raw_inference_service,
+                resource_type="deployment",
+                runtime_name=kueue_kserve_serving_runtime.name,
+            )
+        ]
+        deployments = list(
+            Deployment.get(
+                label_selector=",".join(labels),
+                namespace=kueue_raw_inference_service.namespace,
+                dyn_client=admin_client,
+            )
+        )
+        if len(deployments) != 1:
+            raise DeploymentValidationError("Too many deployments found")
+
+        deployment = deployments[0]
+        deployment.wait_for_replicas(deployed=True)
+        if deployment.instance.spec.replicas != 1:
+            raise DeploymentValidationError("Deployment should have 1 replica")
+
+        # Update inference service to request 2 replicas
+        isvc_to_update = kueue_raw_inference_service.instance.to_dict()
+        isvc_to_update["spec"]["predictor"]["minReplicas"] = 2
+        kueue_raw_inference_service.update(isvc_to_update)
+
+        # Give time for updated deployment
+        time.sleep(10)  # noqa: FCN001
```

Comment:
I've see this pattern used in a few places also on the Vedant PR, it seems that we would ideally have a function like `wait.For(internal, timeout, <condition_lambda_function>)`

It would make the test faster when the condition is reached before the timeout (which is the majority of the cases)